### PR TITLE
LibGUI: Don't update windows that aren't visible.

### PR DESCRIPTION
**Pull request by muscar**
Because the ID of a hidden window is 0, the window server will fail to update them when the system theme is changed. When an application has multiple windows, some of which are hidden, and the system theme is changed (see
https://github.com/SerenityOS/serenity/issues/1378).

This PR changes the window code to ignore update messages if the window has the ID 0--is hidden.

Ideally the window ID would not change, and visibility would be managed separately.

### CODE REVIEW

**Comment by shannonbooth** [comment]
Nothing against your change, definitely an improvement, but I think we should somehow move away from these magic number IDs. They can be a bit hard to follow in some cases for our code. Maybe in the future it would make sense to have a more typesafe ID class, with something like ID::Invalid

**Comment by muscar** [comment]
@shannonbooth I agree. In the short term I changed the code to use `is_visible()` instead of the magic ID number. In the long run we could add strongly typed IDs and change the way window hiding is done. Mutating the ID seems a bit error prone.

**Comment by muscar** [comment]
@awesomekling can you have a look?

I'll add a strongly typed ID type in a separate PR.

**Comment by awesomekling** [comment]
Changes look okay, although please follow the commit message style specified in https://github.com/SerenityOS/serenity/blob/master/CONTRIBUTING.md

Perhaps it would be good to manage visibility separately. What we do currently is deallocate the server-side window object when the client makes a window invisible. Keeping the window ID static would mean an increase in WindowServer memory usage, but maybe it's okay. Why is the static window ID better? What does it solve?

**Comment by muscar** [comment]
@awesomekling I squashed the commits,.

I think decoupling the visibility state and the ID is good. This could be as simple as adding a flag to `LibGUI::Window` and using that instead of the ID to track visibility.

Re. static window IDs: mutating the client side window ID feels a bit odd.

First, because a window can "change identities" after being hidden and shown. If the serenity model is that client window identities are ephemeral then that's fine, and this is a moot point.

Second, because there _may_ be situations where a window zeroes out its ID before it's told to send a message to the window server. In this case it'll send 0 as its ID breaking the window server--what was happening in this case. Of course, you could argue that `update_all_windows()` should have only updated the visible windows in the first place.

We could keep the ID static on the client side and add a way for the client to tell the server it already has a window ID so the server uses that instead of allocating a new one. This way we leave the static server side windows decision for later.

**Comment by muscar** [comment]
Something like this to track the visibility separately: https://github.com/muscar/serenity/commit/09249a054404c5eb472aadbf1a42a2cd71369c0c.

**Comment by awesomekling** [comment]
Sure I guess that would be nice. Let's move forward with your suggestion :)
